app/knight.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+
+class Knight:
+
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            protection: int
+    ) -> None:
+
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = protection
+
+    def adding_value_to_instance(self, element: dict) -> None:
+        self.name = element["name"]
+        self.power = element["power"] + element["weapon"]["power"]
+        self.hp = element["hp"]
+        self.protection = sum(a["protection"] for a in element["armour"])
+
+        if element["potion"] is not None:
+            for value in element["potion"]["effect"]:
+                if value == "power":
+                    self.power += element["potion"]["effect"][value]
+                if value == "hp":
+                    self.hp += element["potion"]["effect"][value]
+                if value == "protection":
+                    self.protection += element["potion"]["effect"][value]

```

Comment:
It is not a good practice to include a method that redefines all attributes at once and later build a logic upon that. Initialize the `Knight` class instance with corresponding values at once and put logic related to filling potion into a separate method